dask/dataframe/io/parquet/arrow.py

```diff
@@ -1285,6 +1287,7 @@ def _collect_file_parts(
                 pa_ds.dataset(
                     files_or_frags,
                     filesystem=fs,
+                    **dataset_options,
```

Comment:
This is the critical change. Without these dataset options, the new fragment may be missing hive/directory-partitioning information.